app/scripts/modules/core/src/pipeline/config/triggers/artifacts/github/github.artifact.ts

```diff
@@ -0,0 +1,42 @@
+import { IController, module } from 'angular';
+
+import { PIPELINE_CONFIG_PROVIDER } from 'core/pipeline/config/pipelineConfigProvider';
+import { IArtifact } from 'core/domain/IArtifact';
+import { PipelineConfigProvider } from 'core/pipeline';
+
+class GcsArtifactController implements IController {
+  constructor(public artifact: IArtifact) {
+    'ngInject';
+    this.artifact.type = 'github/object';
+  }
+}
+
+export const GITHUB_ARTIFACT = 'spinnaker.core.pipeline.trigger.github.artifact';
+module(GITHUB_ARTIFACT, [
+  PIPELINE_CONFIG_PROVIDER,
+]).config((pipelineConfigProvider: PipelineConfigProvider) => {
+  pipelineConfigProvider.registerArtifactKind({
+    label: 'GitHub',
+    description: 'A file stored in git, hosted by GitHub.',
+    key: 'github',
+    controller: 'githubArtifactCtrl',
```

Comment:
You should be able to inline the controller here and skip the class above:
```
controller: (artifact: IArtifact) => { 
  this.artifact = artifact; 
  artifact.type = 'github/object';
}
```
(not 100% sure this will work, but pretty sure)

Reply:
Oh that's so much neater, thanks